eo-maven-plugin/src/main/java/org/eolang/maven/Dependencies.java

```diff
@@ -0,0 +1,178 @@
+            final Collection<Predicate<Dependency>> filters
+        ) {
+            this.dependencies = dependencies;
+            this.filters = filters;
+        }
+
+        @Override
+        public List<Dependency> toList() {
+            return this.dependencies.toList()
+                .stream()
+                .filter(this::applyAllFilters)
+                .collect(Collectors.toList());
+        }
+
+        /**
+         * Apply all filters for Dependency.
+         *
+         * @param dependency Dependency
+         * @return True if all filters were passed
+         */
+        private boolean applyAllFilters(final Dependency dependency) {
+            return this.filters.stream().allMatch(f -> f.test(dependency));
+        }
+    }
+
+    /**
+     * Dependencies uploaded from json file.
+     *
+     * @since 0.28.11
+     */
+    class JsonDependencies implements Dependencies {
+
+        /**
+         * File path.
+         */
+        private final Path file;
+
+        /**
+         * The main constructor.
+         *
+         * @param file File path
+         */
+        JsonDependencies(final Path file) {
+            this.file = file;
+        }
+
+        @Override
+        public List<Dependency> toList() {
+            try {
+                final List<Dependency> all = new ArrayList<>(0);
+                if (Files.exists(this.file)) {
+                    Logger.info(this, String.format("Dependencies file: %s", this.file));
+                    final JsonReader reader = Json.createReader(Files.newBufferedReader(this.file));
+                    final JsonArray artifacts = reader.readObject()
+                        .getJsonArray("artifacts");
+                    for (final JsonValue artifact : artifacts) {
+                        final JsonObject obj = artifact.asJsonObject();
+                        final String group = obj.getString("groupId");
+                        final String id = obj.getString("artifactId");
+                        final String version = obj.getString("version");
+                        final String scope = obj.getJsonArray("scopes").stream()
+                            .map(JsonValue::toString)
+                            .findFirst().orElseThrow(IllegalStateException::new);
+                        all.add(dependency(group, id, version, scope));
+                    }
+                }
+                return all;
+            } catch (final IOException ex) {
+                throw new IllegalStateException(ex);
+            }
+        }
+
+        /**
+         * Factory method.
+         *
+         * @param group Dependency group id
+         * @param artifact Dependency artifact id
+         * @param version Dependency version
+         * @param scope Dependency scope
+         * @return Maven dependency
+         * @checkstyle ParameterNumberCheck (10 lines)
+         */
+        @SuppressWarnings("PMD.UseObjectForClearerAPI")
+        private static Dependency dependency(
```

Comment:
@volodya-lombrozo Can we use ctor here? According to this: https://www.yegor256.com/2017/11/14/static-factory-methods.html

Reply:
@Graur Fixed. Please, take a look. `Dependency` doesn't have a constructor. I've inlined the method.